### PR TITLE
Fix and honor --json flag for find command.

### LINKED ISSUE #2252: mc find doesn't provide json formatted output.
## Expected behaviour
mc find provides json formatted output.

## Actual behaviour
Doesn't provide json formatted output.
```
$ mc find --json s3/
s3/0g3maxon915vry0q3bju206u17qu8jbx/datafile-1-MB
s3/0wx9432ygza5z7wxv33tgax60fl5u3qb/2123123\123
s3/1wjgf35t2n2q8pel6ejatrqf5wv487z9/2123123\123
s3/2fxbw7i9m5oa6bnme2z9o5hcdap069pp/2123123\123
s3/33852c2d0ba7/datafile-1-MB
s3/33852c2d0ba7/datafile-5-MB
s3/33852c2d0ba7/datafile-6-MB
...
...
```

## mc version
master

### PR DESCRIPTION
**Pull request by harshavardhana**
Currently `mc find` didn't honor `--json` flag properly,
this PR addresses that.

Additionally also take care of `List` returning errors
and handle them appropriately.

Fixes #2252

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,4 +9,5 @@ coverage.txt
 parts/
 prime/
 snap/.snapcraft/
-stage/
\ No newline at end of file
+stage/
+mc
```
Comment (balamurugana) [inline .gitignore]: make ending with newline

File: cmd/find.go [hunk 2]
```diff
@@ -43,21 +42,17 @@ import (
 
 // findMSG holds JSON and string values for printing
 type findMSG struct {
```
Comment (balamurugana) [inline cmd/find.go]: I think you have a consistent name like `findMessage`

Comment (harshavardhana) [reply cmd/find.go]: Not part of this PR. Needs a cleanup in separate PR.

Comment (balamurugana) [reply cmd/find.go]: ok fine

File: .gitignore (hunk 1 above)
Comment (harshavardhana) [reply .gitignore]: done

**Pull request merged**